src/list/list-heading.js

```diff
@@ -78,6 +82,9 @@ const ListHeading = React.forwardRef<HeadingPropsT, HTMLLIElement>(
         'endEnhancerDescription will not be rendered if endEnhancer is not a string',
       );
     }
+    if (maxLines && !isMaxLinesValid(maxLines)) {
```

Comment:
Nit: Can we wrap this is a `__DEV__` if statement, maybe do the same for `isEndEnhancerString` check?

Reply:
Good catch, thanks.